source/views/faqs/index.js

```diff
@@ -15,79 +15,62 @@ const styles = StyleSheet.create({
 	},
 })
 
-type Props = {}
+export function FaqView() {
+	let [text, setText] = useState('')
+	let [loading, setLoading] = useState(true)
+	let [refreshing, setRefreshing] = useState(false)
 
-type State = {
-	text: string,
-	loading: boolean,
-	refreshing: boolean,
-}
-
-export class FaqView extends React.PureComponent<Props, State> {
-	static navigationOptions = {
-		title: 'FAQs',
-	}
+	useEffect(() => {
+		fetchData().then(() => setLoading(false))
+	})
 
-	state = {
-		text: '',
-		loading: true,
-		refreshing: false,
-	}
-
-	componentDidMount() {
-		this.fetchData().then(() => {
-			this.setState(() => ({loading: false}))
-		})
-	}
-
-	fetchData = async (reload?: boolean) => {
+	let fetchData = async (reload?: boolean) => {
 		let {text}: {text: string} = await fetch(API('/faqs'), {
 			forReload: reload ? 500 : 0,
 		})
 			.json()
-			.catch(() => ({text: 'There was a problem loading the FAQs'}))
+			.catch(() => setText('There was a problem loading the FAQs'))
 
-		this.setState(() => ({text}))
+		setText(text)
```

Comment:
If you make this return text, instead, this function could be pulled out of the component entirely ;-)

Reply:
I don't understand the advantage of pulling this out, unless you're hinting at centralizing all of the refresh functions across the codebase?

Reply:
Understood.